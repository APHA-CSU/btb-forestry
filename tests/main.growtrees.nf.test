nextflow_process {

    name "Test Process growtrees"
    script "main.nf"
    process "growtrees"

    test("Should generate trees") {

        when {
            params {
            }
            process {
                """
                input[0] = [ 'B6-16', 
                file("${projectDir}/tests/data/snp_only.fas"), 
                file("${projectDir}/accessory/infer_MP_nucleotide_200x.mao"), 
                file("${projectDir}/accessory/analyze_user_tree_MP__nucleotide.mao")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
