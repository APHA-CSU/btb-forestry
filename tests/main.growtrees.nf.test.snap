{
    "Should generate trees": {
        "content": [
            {
                "0": [
                    [
                        "B6-16",
                        "B6-16_22Aug25_MP.nwk:md5,8b553f706124782d4bac73df3e231358"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T15:05:16.211323648"
    }
}